balancer/roundrobin/roundrobin.go

```diff
@@ -70,14 +69,13 @@ type rrPicker struct {
 	// selection from it and return the selected SubConn.
 	subConns []balancer.SubConn
 
-	mu   sync.Mutex
-	next int
+	next uint32
 }
 
 func (p *rrPicker) Pick(balancer.PickInfo) (balancer.PickResult, error) {
-	p.mu.Lock()
-	sc := p.subConns[p.next]
-	p.next = (p.next + 1) % len(p.subConns)
-	p.mu.Unlock()
+	subConnsLen := uint32(len(p.subConns))
+	consumed := atomic.AddUint32(&p.next, 1)
```

Comment:
Can you please rewrite this to similar logic to what we have currently? Right now, this uint32 can overflow, and if it does overflow it can give you weird picks atomic.SwapUint32 could help you here.